src/Compilers/CSharp/Portable/Parser/SlidingTextWindow.cs

```diff
@@ -353,41 +324,13 @@ public char PeekChar(int delta)
         {
             int position = this.Position;
             this.AdvanceChar(delta);
-
-            char ch;
-            if (_offset >= _characterWindowCount
-                && !MoreChars())
-            {
-                ch = InvalidCharacter;
-            }
-            else
-            {
-                // N.B. MoreChars may update the offset.
-                ch = _characterWindow[_offset];
-            }
-
+            var ch = PeekChar();
             this.Reset(position);
             return ch;
         }
```

Comment:
moved to models that defer to existing helpers to avoid duplicated logic and need for all sorts of complexity.